base/ltcounts.dtx

```diff
@@ -328,12 +335,50 @@
        \@latex@warning{Command `\string\the#1' already
                        defined -- not changed}%
      \fi}
+%<latexrelease>\EndIncludeInRelease
+%</2ekernel|latexrelease>
+%<latexrelease>\IncludeInRelease{0000/00/00}{\@definecounter}
+%<latexrelease>                             {provide theHfoo commands}%%
+%<latexrelease>\def\@definecounter#1{\expandafter\newcount\csname c@#1\endcsname
+%<latexrelease>     \setcounter{#1}\z@
+%<latexrelease>     \global\expandafter\let\csname cl@#1\endcsname\@empty
+%<latexrelease>     \@addtoreset{#1}{@ckpt}%
+%<latexrelease>     \global\expandafter\let\csname p@#1\endcsname\@empty
+%<latexrelease>     \expandafter
+%<latexrelease>     \ifx\csname the#1\endcsname\relax
+%<latexrelease>       \expandafter
+%<latexrelease>     \gdef\csname the#1\expandafter\endcsname\expandafter
+%<latexrelease>          {\expandafter\@arabic\csname c@#1\endcsname}%
+%<latexrelease>     \else
+%<latexrelease>       \@latex@warning{Command `\string\the#1' already
+%<latexrelease>                       defined -- not changed}%
+%<latexrelease>     \fi}
+%<latexrelease>\EndIncludeInRelease
+%<*2ekernel>
 %    \end{macrocode}
 %  \end{macro}
 %
 % \begin{macro}{\@addtoreset}
+%  \changes{v1.1o}{2024/09/20}{add the parent theHfoo if a counter is reset}
+%  If a counter is reset when a parent counter changes it no longer has an unique value
+%  across the document. As |\theH<counter>| should be unique 
+%  this representation is changed to include also the 
+%  representation of the parent. 
```

Comment:
Maybe add something like:

This is not 100\% guaranteed to work but has been used this way by hyperref for many years without causing problems.